reactos/drivers/storage/storahci/storahci.c

```diff
@@ -18,38 +18,59 @@
  * @return
  * Return true if intialization was successful
  */
-BOOLEAN AhciPortInitialize(
-  __in      PAHCI_PORT_EXTENSION                portExtension
-)
+BOOLEAN
+AhciPortInitialize (
+    __in      PAHCI_PORT_EXTENSION                portExtension
+    )
 {
     ULONG mappedLength;
     PAHCI_MEMORY_REGISTERS abar;
     PAHCI_ADAPTER_EXTENSION adapterExtension;
     STOR_PHYSICAL_ADDRESS commandListPhysical, receivedFISPhysical;
 
-    StorPortDebugPrint(0, "AhciPortInitialize()\n");
+    DebugPrint("AhciPortInitialize()\n");
+
+    NT_ASSERT(portExtension != NULL);
 
     adapterExtension = portExtension->AdapterExtension;
     abar = adapterExtension->ABAR_Address;
 
+    NT_ASSERT(abar != NULL);
+
     portExtension->Port = &abar->PortList[portExtension->PortNumber];

```

Comment:
Actually AhciPortInitialize routine is being called by only AhciHwFindAdapter, and there I have taken care of all these things
Is is really necessary to NT_ASSERT everything here too?